src/components/overview/OverviewSettingsControl.js

```diff
@@ -18,39 +29,68 @@ const OverviewSettingsControl = ({ otherThresholdPercentage, setOtherThresholdPe
 
     const toolTipFormatter = (value) => `${value}%`;
 
+    const setOtherThresholdPercentageInStore = (value) => {
+        dispatch(setOtherThresholdPercentage(value));
+    };
+
+    const setValueAndCloseModal = () => {
+        setOtherThresholdPercentageInStore(inputValue);
+        setPreviousThreshold(inputValue);
+        writeThresholdToLocalStorage(inputValue);
+        toggleModalVisibility();
+    };
+
+    const cancelModal = () => {
+        setInputValue(previousThreshold);
+        setOtherThresholdPercentageInStore(previousThreshold);
+        toggleModalVisibility();
+    };
+
+    const writeThresholdToLocalStorage = (value) => {
```

Comment:
an alternative I considered here was a redux action creator (not a reducer!) that also handles writes to the store, so that everything is done in the same place. That's probably sensible.

Reply:
I decided against always writing to redux and localStorage at the same time, simply because there's a performance hit with no benefit. At most we only need to write back to localStorage on close ("OK" or "Cancel").